src/main/java/io/boomerang/service/crud/WorkflowServiceImpl.java

```diff
@@ -336,14 +337,25 @@ public WorkflowSummary updateWorkflowProperties(String workflowId,
       userValidationService.validateUserForWorkflow(workflowId);
       entity.setProperties(properties);
       workflowRepository.saveWorkflow(entity);
+      filterPasswordValue(properties);
       return new WorkflowSummary(entity);
     } else {
       entity.setProperties(properties);
       workflowRepository.saveWorkflow(entity);
+      filterPasswordValue(properties);
       return new WorkflowSummary(entity);
     }
   }
 
+  private List<WorkflowProperty> filterPasswordValue (List<WorkflowProperty> properties) {
+	  final String passTypeStr = "password";
+	  if (properties == null) return null;
+	  //If the property is a password, do not return its value, for security reasons.
+		Optional<WorkflowProperty> passProp = properties.stream().filter(f -> passTypeStr.equals(f.getType()) && f.getDefaultValue() != null).findAny();
+		if (passProp.isPresent()) passProp.get().setDefaultValue(null);
```

Comment:
Should this also do the value? or just the default value?

Reply:
As far as I know Workflows use defaultvalue and Teams use value.